File: leet_code/arrays/container_most_water.rb
def max_area(height)
    return nil if !height
    return height[0] if height.length == 1

    left, result = 0, 0
    right = height.length - 1
    while left < right
        temp = [height[left], height[right]].min * (right - left)
        result = temp if temp > result
        height[left] < height[right] ? left += 1 : right -= 1
    end
    result
end

File: leet_code/arrays/container_most_water.js
var maxArea = function(height) {
    if (!height) return null;
    if (height.length === 1) return height[0];

    let left = 0,
        right = height.length - 1,
        result = 0;
    
    while (left < right) {
        let temp = Math.min(height[left], height[right]) * (right - left);
        if (temp > result) result = temp;
        height[left] < height[right] ? left++ : right--;
    }
    return result;
};
